test/scripts/setup-services-run-tests.sh

```diff
@@ -125,28 +131,37 @@ until curl -H "ContentType: application/json" localhost:8080/api/v1alpha1/artifa
     TIMEOUT=$(( TIMEOUT - 1 ))
 done
 
-# Run CURL tests
-cd "${SRC_DIR}/test/e2e" && bash make_requests.sh
-# Test demo notebook
-pip install jupyterlab
-pip install nbconvert
-pip install pandas
-cd "${SRC_DIR}/sdk/python" && \
-  sed -i -e "s@metadata-service.kubeflow:8080@127.0.0.1:8080@" demo.ipynb && \
-  python3 -m nbconvert --to notebook --execute demo.ipynb
-
-# Test resource watcher
-# Port forwarding
+# Port forwarding gRPC server.
 TARGET_GRPC_POD=$(kubectl -n $NAMESPACE get pod -o=name | grep metadata-grpc-deployment | head -1)
 echo "kubectl port-forward from $TARGET_GRPC_POD"
 kubectl -n $NAMESPACE port-forward $TARGET_GRPC_POD 8081:8080 &
 # Stream server logs.
 kubectl -n $NAMESPACE logs -f $TARGET_GRPC_POD &
 
+# Run CURL tests
+cd "${SRC_DIR}/test/e2e" && bash make_requests.sh
+
 # Run Python tests
-pip install ml-metadata==0.15.0
-cd "${SRC_DIR}/sdk/python" && bash tests/run_tests.sh
+cd "${SRC_DIR}/sdk/python"
+rm -rf .testing-env
+virtualenv .testing-env -p /usr/bin/python3.6
+source .testing-env/bin/activate
+python3 -V
+bash tests/run_tests.sh
 
+# Test demo notebook
+pip3 install jupyterlab
+pip3 install nbconvert
+pip3 install pandas
+# Use local server and package.
+sed -i -e "s@metadata-grpc-service.kubeflow@127.0.0.1@" sample/demo.ipynb && \
+sed -i -e "s@grpc_port=8080@grpc_port=8081@" sample/demo.ipynb && \
+sed -i -e "s@pip install kubeflow-metadata --user@pip install -e ${SRC_DIR}/sdk/python@" sample/demo.ipynb && \
+  python3 -m nbconvert --to notebook --execute sample/demo.ipynb
```

Comment:
Seems we cannot ensure the notebook output is consistent with expected? May be better use `papermill.execute_notebook`, we can set the parameters directly without `sed -i` and can assert the outputs.

Reply:
Thanks for pointing papermill. I think it is nice tool to automate the tests with the bash hacks.

You are right we don't check the output string now and it can be improved with papermill. I leave a TODO and want to do it in next PR.